pkg/datastore/test/namespace.go

```diff
@@ -175,3 +180,49 @@ func EmptyNamespaceDeleteTest(t *testing.T, tester DatastoreTester) {
 	_, _, err = ds.SnapshotReader(deletedRev).ReadNamespace(ctx, testfixtures.UserNS.Name)
 	require.True(errors.As(err, &datastore.ErrNamespaceNotFound{}))
 }
+
+// StableNamespaceReadWriteTest tests writing a namespace to the datastore and reading it back,
+// ensuring that it does not change in any way and that the deserialized data matches that stored.
+func StableNamespaceReadWriteTest(t *testing.T, tester DatastoreTester) {
+	require := require.New(t)
+
+	schemaString := `definition document {
+	relation viewer: user | user:*
+	relation editor: user | group#member
+	relation parent: organization
+	permission edit = editor
+	permission view = viewer + edit + parent->view
+	permission other = viewer - edit
+	permission intersect = viewer & edit
+	permission with_nil = ((viewer - edit) & parent->view) & nil
+}`
```

Comment:
this could be table-driven with a few different schemas

Reply:
Could be, but this schema basically encompasses the entire set of operations we currently support. Once we extend schema further, I'd be fine with making this table driven